packages/admin-ui/client/pages/Item/index.js

```diff
@@ -112,18 +130,17 @@ const ItemDetails = withRouter(
         item: initialData,
       } = this.props;
 
-      resolveAllKeys(
+      Promise.all([
+        getInitialValues(fields, initialData),
+        getCurrentValues(fields, item),
+      ]).then(([initialValues, currentValues]) => (
         // Don't try to update anything that hasn't changed.
         // This is particularly important for access control where a field
         // may be `read: true, update: false`, so will appear in the item
         // details, but is not editable, and would cause an error if a value
         // was sent as part of the update query.
-        arrayToObject(
-          fields.filter(field => !isEqual(field.getValue(initialData), field.getValue(item))),
```

Comment:
calling `field.getValue()` in the `.filter()` call could potentially return a Promise, so the `isEqual()` would be comparing a `new Promise()` vs `new Promise()` which will never be equal.